src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -0,0 +1,169 @@
+package dotty.tools.dotc
+package transform
+
+import dotty.tools.dotc.transform.TreeTransforms.{TransformerInfo, TreeTransform, TreeTransformer}
+import dotty.tools.dotc.ast.{Trees, tpd}
+import scala.collection.{ mutable, immutable }
+import ValueClasses._
+import scala.annotation.tailrec
+import core._
+import typer.ErrorReporting._
+import typer.Checking
+import Types._, Contexts._, Constants._, Names._, NameOps._, Flags._, DenotTransformers._
+import SymDenotations._, Symbols._, StdNames._, Annotations._, Trees._, Scopes._, Denotations._
+import util.Positions._
+import Decorators._
+import Symbols._, TypeUtils._
+
+/** A macro transform that runs immediately after typer and that performs the following functions:
```

Comment:
@vladimirnik Here is a list of everything PostTyper does. To duplicate from scalac I believe you need to do
everything in (1). Alternatively, try to run scalac's existing SuperAccessors before PatternMacther. I am not sure the 
comment that says you can't is still accurate. (2), (3) and (4) are done in  scalac's typer already, so no need to redo.
I am not sure where (5)-(8) are done; you'd have to check that and possibly duplicate. But they are small things.